### ISSUE TITLE
Metric generator generates tags from constant and static fields

### ISSUE THREAD

**Issue by adamnova** [dotnet/extensions#4787]
### Description

Metrics generator generates tags (dimensions) from constant and static variables. This is not ideal behaviour as constants can be useful for something like this `[Histogram(typeof(TestMetric), Name = TestMetric.Name)]` where the Name is inside the `TestMetric` struct.

### Reproduction Steps

```
internal static partial class Metrics
{
    [Histogram(typeof(TestMetric), Name = "MyCustomMetricName")]
    public static partial SomeMetric CreateMetric(Meter meter);
}

public struct TestMetric()
{
    public const string TestConstant = "a";
    public static string TestStatic = "b";
    public string TestNormal = "c";
}
```

### Expected behavior

```
var tagList = new global::System.Diagnostics.TagList
{
    new global::System.Collections.Generic.KeyValuePair<string, object?>("TestNormal", o.TestNormal!),
};
```

### Actual behavior

```
var tagList = new global::System.Diagnostics.TagList
{
    new global::System.Collections.Generic.KeyValuePair<string, object?>("TestConstant", o.TestConstant!), // cannot be accessed by instance
    new global::System.Collections.Generic.KeyValuePair<string, object?>("TestStatic", o.TestStatic!), // cannot be accessed by instance
    new global::System.Collections.Generic.KeyValuePair<string, object?>("TestNormal", o.TestNormal!),
};
```

### Known Workarounds

* Do not use statics or constants inside metric data types.
* Create inner static class `Constants` and put constants there.

**Comment by xakep139** [comment]
@klauco thoughts?

**Comment by xakep139** [comment]
Thanks for reporting this @adamnova! The fix should be released in 8.1
